lib/pbench/server/api/resources/users_api.py

```diff
@@ -387,59 +419,60 @@ def put(self, username):
                         "message": "failure message"
                     }
         """
-        post_data = request.get_json()
-        if not post_data:
+        user_payload = request.get_json()
+        if not user_payload:
             self.logger.warning("Invalid json object: {}", request.url)
-            abort(400, message="Invalid json object in request")
+            abort(HTTPStatus.BAD_REQUEST, message="Invalid json object in request")
 
         try:
-            user, verified = self.auth.verify_user(username)
-        except Exception:
-            self.logger.exception("Exception occurred while verifying the user")
-            abort(500, message="INTERNAL ERROR")
-
-        # TODO: Check if the user has the right privileges
-        if not verified:
-            self.logger.warning(
-                "User {} is not authorized to delete user {}.", user.username, username,
+            target_user, http_status, http_message = self.get_valid_target_user(
+                target_username, "PUT"
             )
-            abort(
-                403,
-                message=f"Not authorized to update information about user {username}",
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in PUT request from user {!r} while validating target user {!r}",
+                self.auth.token_auth.current_user().username,
+                target_username,
             )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        if not target_user:
+            abort(http_status, message=http_message)
 
         # Check if the user payload contain fields that are either protected or
         # are not present in the user db. If any key in the payload does not match
         # with the column name we will abort the update request.
-        non_existent = set(post_data.keys()).difference(
+        non_existent = set(user_payload.keys()).difference(
             set(User.__table__.columns.keys())
         )
         if non_existent:
             self.logger.warning(
                 "User trying to update fields that are not present in the user database. Fields: {}",
                 non_existent,
             )
-            abort(400, message="Invalid fields in update request payload")
-        protected = set(post_data.keys()).intersection(set(User.get_protected()))
+            abort(
+                HTTPStatus.BAD_REQUEST,
+                message="Invalid fields in update request payload",
+            )
+        protected = set(user_payload.keys()).intersection(set(User.get_protected()))
```

Comment:
Unless the current user `is_admin()` **and** the target user is different from the current user, then I think `role` should be added to the list of protected fields.  (That is, an admin user can change _another_ user's role; otherwise, users cannot change their own roles or the roles of others.)

Reply:
And, I think that this is the appropriate place in the code to implement this policy.